roc/version.go

```diff
@@ -72,3 +79,43 @@ func parseVersion(s string) SemanticVersion {
 
 	return v
 }
+
+// Validate version compatibility of roc bindings and native library.
+// This function must be called at all library entry point atleast once.
+// Entry points refer to exported non-method functions of this package
+// This includes OpenContext, OpenSender, OpenReceiver, ParseEndpoint,
+// SetLogger, SetLoggerFunc and SetLogLevel
+func (version Versions) Validate() error {
```

Comment:
I think this part:

```
// This function must be called at all library entry point atleast once.
// Entry points refer to exported non-method functions of this package
// This includes OpenContext, OpenSender, OpenReceiver, ParseEndpoint,
// SetLogger, SetLoggerFunc and SetLogLevel
```

Should be moved to versionCheck.

Here, in Versions.Validate, comment is for the user, and user is not responsible for calling Validate in entrypoints and does not have to know about it.

In Versions.Validate comment it's better to say something like: "if versions are incompatible, will return error describing the problem, otherwise will return nil".

Also I think we should not list all entrypoints here. We already have one list in TestVersion_Entrypoints. We can easily forget to update list in comment and it will become outdated. 

Also "atleast" => "at least".